src/components/SearchModal/sorting.ts

```diff
@@ -39,14 +42,17 @@ function getTokenComparator(balances: {
 }
 
 export function useTokenComparator(inverted: boolean): (tokenA: Token, tokenB: Token) => number {
+  // todo namgold: optimize this
   const balances = useAllTokenBalances()
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  const comparator = useMemo(() => getTokenComparator(balances ?? {}), [JSON.stringify(balances)])
-  return useMemo(() => {
+  const debouncedBalances = useDebounce(balances, 500)
+  const comparator = useMemo(() => getTokenComparator(debouncedBalances ?? {}), [debouncedBalances])
```

Comment:
Put comparator inside result's useMemo? as it's used nowhere else.
We then reduce 2 useMemo into just one